edweiss-app/components/courses/MaterialComponent.tsx

```diff
@@ -87,63 +143,98 @@ interface MaterialProps {
  * @param onDelete - The function to be called when the user deletes the material.
  * @param material - The material to be edited.
  * 
  * 
  * @returns JSX.Element - The rendered component for the material creation inner-page.
  */
-const MaterialComponent: ReactComponent<MaterialProps> = ({ mode, onSubmit, onDelete, material }) => {
+const MaterialComponent: ReactComponent<MaterialProps> = ({ mode, courseId, onSubmit, onDelete, material }) => {
 
     const [title, setTitle] = useState<string>(material?.data.title ?? "");
     const [description, setDescription] = useState<string>(material?.data.description ?? "");
     const [fromDate, setFromDate] = useState<Date>(material ? Time.toDate(material.data.from) : new Date());
     const [toDate, setToDate] = useState<Date>(material ? Time.toDate(material.data.to) : new Date());
+    const [docs, setDocs] = useState<MaterialDocument[]>(material?.data.docs ?? []);
+    const [docTitle, setDocTitle] = useState<string>('');
+    const [docType, setDocType] = useState<MaterialType>('slide');
+    const [docName, setDocName] = useState<string>('');
+    const [docData, setDocData] = useState<string>('');
+    const [documentsData, setDocumentsData] = useState<Array<{ uri: string, dataBase64: string }>>([]);
+    const [docFormat, setDocFormat] = useState<string>('');
+    const [deletedDocs, setDeletedDocs] = useState<MaterialDocument[]>([]);
     const [titleChanged, setTitleChanged] = useState<boolean>(false);
     const [fromDateChanged, setFromDateChanged] = useState<boolean>(false);
     const [fromTimeChanged, setFromTimeChanged] = useState<boolean>(false);
     const [toDateChanged, setToDateChanged] = useState<boolean>(false);
     const [toTimeChanged, setToTimeChanged] = useState<boolean>(false);
+    const [docTitleChanged, setDocTitleChanged] = useState<boolean>(false);
     const [showPickerFromDate, setShowPickerFromDate] = useState<boolean>(false);
     const [showPickerFromTime, setShowPickerFromTime] = useState<boolean>(false);
     const [showPickerToDate, setShowPickerToDate] = useState<boolean>(false);
     const [showPickerToTime, setShowPickerToTime] = useState<boolean>(false);
-
+    const [showAddDocument, setShowAddDocument] = useState<boolean>(false);
     const [isButtonDisabled, setIsButtonDisabled] = useState(true);
+    const [isDocumentButtonDisabled, setIsDocumentButtonDisabled] = useState(true);
+    const [documentFieldsNoneEmpty, setDocumentFieldsNoneEmpty] = useState(false);
+
+    const resetDocument = () => {
+        setDocTitle('');
+        setDocType('slide');
+        setDocName('');
+        setDocData('');
+        setDocTitleChanged(false);
+    }
 
+    //===============================================================
+    //======================== Listeners ============================
+    //===============================================================
+
+    // Document Format Listener
+    useEffect(() => {
+        if (docName) {
```

Comment:
you wrote if(string) : I do not understand this syntax, maybe consider putting a boolean instead.

Reply:
If docName is empty, null or undefined it will not be evaluated as true. That is why I write it like this